### ISSUE TITLE
Computation of Minimum,Mean and Maximum from hourly data

### ISSUE THREAD

**Issue by mhabimana** [climsoft/Climsoft#645]
A feature for computation of Minimum,Mean (average) and Maximum from hourly data has been requested by both @fmucyo (Rwanda) and @berenger07 (Cote d'Ivoire).

**Comment by isedwards** [comment]
@dannyparsons - can we get your thoughts on doing this in Climsoft 4 using R (possibly without having to install the whole of R-Instat)?

**Comment by mhabimana** [comment]
@isedwards - Thanks
I am thinking some more radio button may be added under **Summary Type** to select **Maximum** or **Minimum** computation or calculation.
![image](https://user-images.githubusercontent.com/10340170/110086901-02cc8b00-7da4-11eb-926c-3608ddaf515f.png)

**Comment by smachua** [comment]
@mhabimana Climsoft V4 can provide these products although not in one go.

(i) The Mean of Hourly data will be produced when you select products for Daily, Monthly or Annual. Usually those elements that are observed several times in a day will be averaged according to the selected duration (i.e. daily, monthly or annual). Even the minutes observations say from AWS can be averaged through the product Hourly.

(ii) Minimum and maximum values will be produced through Extremes Product. See the  following dialog:
![image](https://user-images.githubusercontent.com/12034121/110089201-dbc38880-7da6-11eb-921a-fdd6e3f1d4d2.png)

**Comment by mhabimana** [comment]
@smachua - Thanks,This is noted.
@fmucyo and @berenger07 : can you try this and give us your feedback?

**Comment by berenger07** [comment]
OK I'll get back to you shortly.


KOUACOU Adou Kouassi Doré BerengerIngénieur en Télédétection, SIG et Bases de Données Direction de la météorologie Nationale (SODEXAM)Contact: +225 0708736793/+225 0505655191Boîte Postale: 10 BP 3310 Abidjan 10

 

    Le vendredi 5 mars 2021, 09:15:48 UTC, Marcellin <notifications@github.com> a écrit :  
 
 


@smachua - Thanks,This is noted.
@fmucyo and @berenger07 : can you try this and give us your feedback?

**Comment by mhabimana** [comment]
@berenger07,@fmucyo: Daily extremes feature will in Climsoft will be able to compute fro you daily Min and Max from Hourly data if you select the right element for hourly data (e.g: Dry bulb(code:101),see below:
![image](https://user-images.githubusercontent.com/10340170/110113937-eb51ca00-7dc4-11eb-9a49-b517d9080ce3.png)

![image](https://user-images.githubusercontent.com/10340170/110113957-f3aa0500-7dc4-11eb-9ad7-1240894a949e.png)

**Comment by berenger07** [comment]
Bonjour ,
I get this message when I run an update after correcting and saving the file by appending updated to the lower and upper limit file name


cannot add or update a child row a foreign key constraint fails (mariadb climsoft_db_v4, observation initial , CONSTRAINT 'FK_maria_climsoft_db_v4_station_observationinitial' FOREIGN KEY (recorderForm) REFERENCES 'stations' ('stationId))

ci-joint l'image de la capture (attached the image of the capture)
need a solution to get there

Cordialement
KOUACOU Adou Kouassi Doré BerengerIngénieur en Télédétection, SIG et Bases de Données Direction de la météorologie Nationale (SODEXAM)Contact: +225 0708736793/+225 0505655191Boîte Postale: 10 BP 3310 Abidjan 10

**Comment by mhabimana** [comment]
@berenger07 - I can not see the screenshot,please,attached this. Are you trying to do Quality Control(QC) of data?

**Comment by berenger07** [comment]
Yes I try to do the quality control. But I get an error message. 
I am attaching the message to you in a document.
and another concern. Is it possible to put the interface in French.
Thank you
KOUACOU Adou Kouassi Doré BerengerIngénieur en Télédétection, SIG et Bases de Données Direction de la météorologie Nationale (SODEXAM)Contact: +225 0708736793/+225 0505655191Boîte Postale: 10 BP 3310 Abidjan 10

 

    Le mardi 27 avril 2021, 14:25:40 UTC, Marcellin ***@***.***> a écrit :  
 
 


@berenger07 - I can not see the screenshot,please,attached this. Are you trying to do Quality Control(QC) of data?

**Comment by mhabimana** [comment]
@berenger07 - Noted but still I am not seeing any attachment. Can you respond via GitHub and use the GitHub facility at the bottom of the page to attach the screenshot. 
Regarding the French interface,this is not yet available at the moment but we have the plan to translate the software in other languages,including French.

**Comment by berenger07** [comment]
attached the capture of the error message 
![egb](https://user-images.githubusercontent.com/36269507/116262311-f2171080-a767-11eb-895f-f43dd5121a81.jpg)

**Comment by berenger07** [comment]
no, I haven't changed anything, I just correct the values ​​below and above the limits I had defined and I rename the file by adding updated. I enclose the files of the HQ
[qc_report_lowerlimit_202101_202101.csv](https://github.com/climsoft/Climsoft/files/6385622/qc_report_lowerlimit_202101_202101.csv)
[qc_report_lowerlimit_202101_202101_updated.csv](https://github.com/climsoft/Climsoft/files/6385623/qc_report_lowerlimit_202101_202101_updated.csv)
[qc_report_upperlimit_202101_202101.csv](https://github.com/climsoft/Climsoft/files/6385624/qc_report_upperlimit_202101_202101.csv)
[qc_report_upperlimit_202101_202101_updated.csv](https://github.com/climsoft/Climsoft/files/6385625/qc_report_upperlimit_202101_202101_updated.csv)

**Comment by mhabimana** [comment]
@berenger07 -Thanks for sharing the QC files report,will look at it and come back to you.

**Comment by berenger07** [comment]
Ok thank you I am waiting.

KOUACOU Adou Kouassi Doré BerengerIngénieur en Télédétection, SIG et Bases de Données Direction de la météorologie Nationale (SODEXAM)Contact: +225 0708736793/+225 0505655191Boîte Postale: 10 BP 3310 Abidjan 10

 

    Le mercredi 28 avril 2021, 07:29:53 UTC, Marcellin ***@***.***> a écrit :  
 
 


@berenger07 -Thanks for sharing the QC files report,will look at it and come back to you.

**Comment by mhabimana** [comment]
@berenger07 - I am trying to reach you via what'sup but it seems you are not connected.

**Comment by berenger07** [comment]
Ok I am connected now.

**Comment by mhabimana** [comment]
@berenger07 -I am using your number ending with 93. Is it the one connected to what's up?

**Comment by berenger07** [comment]
Yes it's the number that passes
